hashintel/hash
# Local kratos email verification with mailslurper

## 🌟 What is the purpose of this PR?



This PR adds email verification flows to Kratos, configuring the `code` method and custom mail templates.
There are some potential type issues with the kratos SDK for typescript, see https://github.com/ory/kratos/issues/2943 which means the current implementation casts a type in a place where it isn't expected.

The current setup allows for verification through the UI, but it isn't neatly integreated/expected. I've included the UI scaffold for prettifying

## 🔗 Related links





- [Asana task](https://app.asana.com/0/1203363157432081/1203509372806154/f) _(internal)_

## 🚫 Blocked by





## 🔍 What does this change?




- Kratos dev config (and anticipation for prod config)
- Adds new frontend endpoint to receive verification tokens
- Email sending through mailslurper locally
- Email templates for HTML and plaintext

## 📜 Does this require a change to the docs?




## ⚠️ Known issues


- The typescript types on the frontend for submitting the verification flow seems to be incorrect, hence some casting to get it to work.
- Production does not have this enabled yet

## 🐾 Next steps


- Prettier email templates
- Enabling _n production_

## 🛡 What tests cover this?

none

## ❓ How to test this?



1.  Checkout the branch as you would normally (external-service + yarn dev)
1.  Sign up a new user and choose a shortname/preferred name
1.  Go to http://localhost:4436/# to access mailslurper
1.  open network explorer tab in dev tools
1.  Click verification link and verify email
1.  Observe next `whoami` request contains a verified mail

## 📹 Demo



https://user-images.githubusercontent.com/6988668/207086462-87f256d3-b6ca-441d-9d1c-e7ddbba89222.mp4
